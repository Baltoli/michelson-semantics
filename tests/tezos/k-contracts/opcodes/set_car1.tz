$contract { parameter string;
storage (pair (string %s) (nat %n));
code { DUP; CDR; DIP{CAR}; SET_CAR %s; NIL operation; PAIR };
 } ;
$param "abc" ;
$storage (Pair "hello" 0) ;
$ExpectedOutput (Pair "abc" 0) ;

