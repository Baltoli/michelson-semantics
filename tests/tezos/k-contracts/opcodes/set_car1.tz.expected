<generatedTop>
  <k>
    SET_CAR %s  .AnnotationList ~> NIL .AnnotationList operation .AnnotationList ; PAIR .AnnotationList ~> #TransformExpected ( Pair "abc" 0 )
  </k>
  <stack>
    Pair "hello" 0 ~> "abc"
  </stack>
  <paramtype>
    string .AnnotationList
  </paramtype>
  <storagetype>
    pair .AnnotationList string %s  .AnnotationList nat %n  .AnnotationList
  </storagetype>
  <balance>
    #Mutez ( 0 )
  </balance>
  <amount>
    #Mutez ( 0 )
  </amount>
  <now>
    #Timestamp ( -1 )
  </now>
  <myaddr>
    #Address ( "InvalidMyAddr" )
  </myaddr>
  <knownaddrs>
    .Map
  </knownaddrs>
  <sourceaddr>
    #Address ( "InvalidSourceAddr" )
  </sourceaddr>
  <senderaddr>
    #Address ( "InvalidSenderAddr" )
  </senderaddr>
  <chainid>
    #ChainId ( 0 )
  </chainid>
</generatedTop>
